consensus/istanbul/core/handler.go

```diff
@@ -121,11 +121,17 @@ func (c *core) handleEvents() {
 					//c.backend.Gossip(c.valSet, p)
 				}
 			}
-		case _, ok := <-c.timeoutSub.Chan():
-			if !ok {
+		case ev, ok := <-c.timeoutSub.Chan():
+			if !ok || ev.Data == nil {
+				logger.Warn("Invalid message type from timeout channel")
+				return
+			}
+			data, ok := ev.Data.(timeoutEvent)
+			if !ok || data.view == nil {
+				logger.Warn("Invalid message from timeout channel", "msg", ev.Data)
 				return
 			}
```

Comment:
Due to this code, we should upgrade all CN at the same time. Could we just accept `type timeoutEvent struct {
	nextView *istanbul.View
}` and `type timeoutEvent struct {}` at the same time for now? After all CN has upgraded and started sending the new struct of `timeoutEvnet`, we can change this logic to handle only the new struct

Reply:
@winnie-byun TimeoutEvent is not propagated to other nodes. It is a local event and is sent to the node itself. Therefore, we don't need to consider version matching among CNs.